agent/sighandlers/termination_handler.go

```diff
@@ -98,3 +100,37 @@ func FinalSave(saver statemanager.Saver, taskEngine engine.TaskEngine) error {
 	}
 	return nil
 }
+
+func saveStateAll(state dockerstate.TaskEngineState, dataClient data.Client) {
+	// save all tasks state data
+	for _, task := range state.AllTasks() {
+		if err := dataClient.SaveTask(task); err != nil {
+			seelog.Errorf("Failed to save data for task %s: %v", task.Arn, err)
+		}
+	}
+
+	// save all container and docker container state data
+	for _, containerId := range state.GetAllContainerIDs() {
+		container, ok := state.ContainerByID(containerId)
+		if !ok {
+			seelog.Errorf("Unable to find container for %s", containerId)
+		}
+		if err := dataClient.SaveDockerContainer(container); err != nil {
+			seelog.Errorf("Failed to save data for docker container %s: %v", container.Container.Name, err)
+		}
+	}
+
+	// save all image state data
+	for _, imageState := range state.AllImageStates() {
+		if err := dataClient.SaveImageState(imageState); err != nil {
+			seelog.Errorf("Failed to save data for image state: %s, %v", imageState.GetImageID(), err)
```

Comment:
nit: would suggest use the same logging format: `for image state %s: %v`

Reply:
Thanks for pointing it out. Will fix it.